spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/cloudfoundry/GcpCloudFoundryEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.core.cloudfoundry;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.config.ConfigFileApplicationListener;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.boot.json.JsonParser;
+import org.springframework.boot.json.JsonParserFactory;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.util.StringUtils;
+
+/**
+ * Converts GCP service broker metadata into Spring Cloud GCP configuration properties.
+ *
+ * @author João André Martins
+ */
+public class GcpCloudFoundryEnvironmentPostProcessor
+		implements EnvironmentPostProcessor, Ordered {
+
+	public static final String VCAP_SERVICES_ENVVAR = "VCAP_SERVICES";
+
+	private static final Log LOGGER =
+			LogFactory.getLog(GcpCloudFoundryEnvironmentPostProcessor.class);
+
+	private static final String SPRING_CLOUD_GCP_PROPERTY_PREFIX = "spring.cloud.gcp.";
+
+	private JsonParser parser = JsonParserFactory.getJsonParser();
+
+	private int order = ConfigFileApplicationListener.DEFAULT_ORDER - 1;
+
+	private enum GcpCfService {
+
+		pubsub("google-pubsub",
+				ImmutableMap.of("ProjectId", "project-id",
+				"PrivateKeyData", "credentials.encodedKey",
```

Comment:
Shouldn't it be like `credentials.encoded-key` for consistency with all other *lower-case-dashed* ?

I've been told recently that Spring Boot `@ConditionalOnProperty` doesn't support relaxed values anymore.
It is now only like this:
```
	 * Use the dashed notation to specify each property, that is all lower case with a "-"
	 * to separate words (e.g. {@code my-long-property}).
	 * @return the names
	 */
	String[] name() default {};
```

Reply:
Good catch! I thought I had gone this way since the beginning for encoded key...